independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/api/WebClientApplicationException.java

```diff
@@ -20,22 +20,27 @@
 import org.jboss.resteasy.reactive.common.jaxrs.StatusTypeImpl;
 
 /**
- * Subclass of {@link WebApplicationException} for use by clients, which forbids setting a response that
- * would be used by the server.
- * FIXME: I'd rather this be disjoint from WebApplicationException, so we could store the response info
- * for client usage. Perhaps we can store it in an alternate field?
```

Comment:
This explains that we did not want the client response to end up in the `WebApplicationException.response` field, and you're doing exactly that.
Doing this, we run the risk of user exception mappers that turn that `Response` into a server response, which is a huge security issue.
I'd much rather this response get added as a new field property `getClientReponse()`, would this work for you?

Reply:
I was aware of this, but I decided to put it in the response field anyway because now the exception implements `ResteasyReactiveClientProblem` to prevent that. Having another field to hold the response may result confusing, but I can add that property.

Reply:
It is not ideal, but the client web application exception type should be disjoint to the server web application exception type in an ideal world.

I think the (client) `Response` should go into another property, to make sure nobody ever treats this as a server response. The security risk is too great, in my opinion.